### PR TITLE
Always allow key creation where less than 2 exist

### PR DESCRIPTION
**Pull request by leigh-hogarth**

### CODE REVIEW

**Comment by kneemaa** [comment]
@leigh-hogarth What is the use case for this? In what scenario would the repository have 0 keys and would update itself to have keys?

**Comment by leigh-hogarth** [comment]
Hi @kneemaa - there are instances where an IAM user can be created without any pre-existing access keys.
For example, if you were to create an IAM user via a tool such as Terraform, due to key rotation you wouldn't want to create the access keys using the same tool. Therefore the result is an initial empty user.
This action in its original state will then fail unless you manually create an initial set of keys, which is why I created this PR to allow for the use-case and avoid needing manual intervention.

**Comment by kneemaa** [comment]
@leigh-hogarth - if the user has no keys, how will it's first set of keys be set in Github?

**Comment by leigh-hogarth** [comment]
@kneemaa they won't be. This change allows you start from scratch with an empty IAM user and empty GH secrets.

**Comment by kneemaa** [comment]
@leigh-hogarth I do not see what is resolving. 

The Action will not work until Keys are set and accessible.

Having this action run when there are no keys is not desired, having it fail when no keys are found is a way to alert that your other actions will fail because there are no valid keys.

**Comment by leigh-hogarth** [comment]
@kneemaa the use case is there for setting things up from scratch.
The action _does_ work without those keys being present as we actually use it in a central workflow which manages these keys on other repositories.
Therefore so long as that central workflow has AWS access, it can run this action against other repositories.
Those other repositories may be in an infant state, so will not have keys set as secrets on first run, hence why this update allows for that.

**Comment by kneemaa** [comment]
@leigh-hogarth Thanks for the clarification, could you please update the README to account for this and how it works?

**Comment by leigh-hogarth** [comment]
@kneemaa README updated.

File: rotate_keys.py [hunk 1]
```diff
@@ -26,15 +26,12 @@ def main_function():
...
-
     # Check if two keys already exist, if so, exit 1
-    if starting_num_keys != 1:
-        print("There are already 2 keys for this user, Cannot rotate tokens")
+    if starting_num_keys >= 2:
+        print("There are already 2 keys for this user. Cannot rotate tokens.")
         sys.exit(1)
     else:
-        print(f"I have {starting_num_keys} token, proceeding.")
+        print(f"Validated <2 keys exist ({starting_num_keys}), proceeding.")
```
Comment (kneemaa) [inline rotate_keys.py:34]: This is going to print as `Validated <2 keys exist 0, proceeding.` or `Validated <2 keys exist 1, proceeding.`

Suggesting the print be updated to have `{starting_num_keys}` removed or the output to be clear

File: README.md [hunk 2]
```diff
@@ -3,6 +3,14 @@
 
 # Rotate AWS Access token stored in Github Repository secrets
 
+Performs the following actions:
+1. Checks for existing IAM access and secret key pairs of the provided IAM user (`IAM_USERNAME`)
+1. If 2 sets of keys exists, the action will fail
+1. If 0 or 1 set of keys exists, the action will:
```
Comment (kneemaa) [inline README.md:9]: This needs to be clear that having 0 keys will not work if the action is run in the Target repository (original functionality)

https://github.com/kneemaa/github-action-rotate-aws-secrets/pull/22#issuecomment-1060756380

**Review by kneemaa — CHANGES_REQUESTED** [review]
two requests

File: rotate_keys.py (hunk 1 above)
Comment (leigh-hogarth) [reply rotate_keys.py]: Updated the text in the latest commit

File: README.md (hunk 2 above)
Comment (leigh-hogarth) [reply README.md:9]: This is irrelevant. The action won't run without appropriate access keys on the repository and never would.

**Pull request merged**